sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -942,21 +945,21 @@ def get_manifest(self, repository: str, tag_or_digest: str, **kwargs) -> GetMani
                 **kwargs
             )
         )
+        if response.http_response.headers['Content-Length'] > MAX_MANIFEST_SIZE:
+            raise ValueError("Manifest size is bigger than 4MB.")
```

Comment:
What was the error before? Is this a breaking change?

Reply:
It's not a breaking change, we didn't check the size before. It should be able to download successfully before.

Reply:
Just checking my understanding - without the added validation in the PR, this would download successfully before? Why are we adding the validation?

Reply:
Yes, it was downloading successfully. We changed to add manifest size validation following [contained](https://github.com/containerd/containerd/blob/main/remotes/docker/resolver.go#L59) motivation and to protect from malicious unbounded responses.

Reply:
Ah okay, so this is a true GA breaking change made for security reasons. Did this get approved by architects already?

Reply:
It's a decision made across languages, no apiview updates with this updates, I don't know if Anna has approved. cc: @annatisch to help.

Reply:
If this change is being made across languages - there should be an email with the archboard about this?

Reply:
This is consistent with behavior of tooling in the ecosystem, e.g. containerd.  https://github.com/Azure/azure-sdk-for-net/issues/35771

Reply:
> Is it possible to have a malicious manifest < 4MB?

Yes, but the goal is not to protect from malicious content, but to fail fast. Small manifests are fine, but if they are too big,  we won't spend resources in client applications attempting to download, buffer and calculate digest of invalid data a.

Reply:
Thanks for the additional context!
If there's no valid manifests >4MB that would have previously been working then this should be fine :)